lumodon_questions.md

```diff
@@ -78,6 +78,255 @@ var nameOfFirstCar = _.compose( _.head, _.prop('name') );
 ```
 It doesn't pass.
 
+====================== Notes ======================
 ## Important Rule:
 // map's composition law
-var law = compose(map(f), map(g)) === map(compose(f, g));
\ No newline at end of file
+var law = compose(map(f), map(g)) === map(compose(f, g));
+
+## Exercise 4 from functors
+Note: This line from chapter 8 functors exercise makes so much sense! You're piping information from one function to another. Not sure why this only just now makes sense.
+```
+var ex4 = _.compose( Maybe.of, parseInt );
+```
+
+==================== Code Wars =====================
+# Code Wars Exercises
+## Exercise 1: Calculate average
+
+My Solution:
+```
+function find_average(array) {
+  let r = array.reduce( (a,b) => {
+    return a + b
+  })
+  
+  return r / array.length
+}
+```
+
+Best Solution (as voted on by others):
+```
+function find_average(array) {
+  return array.reduce((result, current) => result + current, 0) / array.length;
+}
+```
+
+## Exercise 2: Multiply All
+
+My Solution:
+```
+const multiplyAll = array => {
+  return multiplicand => {
+    const resultArray = []
```

Comment:
Thank you. I'll have to get more familure with those, good to know what the pattern is to look out for so that I can identify procedurally when I should be thinking of another way.